api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/MetricHandler.java

```diff
@@ -70,33 +71,32 @@
     @GET
     @Path("/")
     @ApiOperation(value = "Find tenant's metric definitions.", notes = "Does not include any metric values. ",
-            response = List.class, responseContainer = "List")
-    @ApiResponses(value = { @ApiResponse(code = 200, message = "Successfully retrieved at least one metric "
-            + "definition."),
-            @ApiResponse(code = 204, message = "No metrics found."),
-            @ApiResponse(code = 400, message = "Given type is not a valid type.", response = ApiError.class),
+                  response = List.class, responseContainer = "List")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully retrieved at least one metric definition.",
+                         response = ApiError.class),
+            @ApiResponse(code = 204, message = "No metrics found.",
+                         response = ApiError.class),
```

Comment:
I assumed it meant no response body but wasn’t 100% certain. Thanks for the clarification.

> On Jun 22, 2015, at 2:31 PM, Thomas Segismont <notifications@github.com> wrote:
> 
> In api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/MetricHandler.java <https://github.com/hawkular/hawkular-metrics/pull/270#discussion_r32967276>:
> 
> > @@ -70,33 +71,32 @@
> >      @GET
> >      @Path("/")
> >      @ApiOperation(value = "Find tenant's metric definitions.", notes = "Does not include any metric values. ",
> > -            response = List.class, responseContainer = "List")
> > -    @ApiResponses(value = { @ApiResponse(code = 200, message = "Successfully retrieved at least one metric "
> > -            + "definition."),
> > -            @ApiResponse(code = 204, message = "No metrics found."),
> > -            @ApiResponse(code = 400, message = "Given type is not a valid type.", response = ApiError.class),
> > +                  response = List.class, responseContainer = "List")
> > +    @ApiResponses(value = {
> > +            @ApiResponse(code = 200, message = "Successfully retrieved at least one metric definition.",
> > +                         response = ApiError.class),
> > +            @ApiResponse(code = 204, message = "No metrics found.",
> > +                         response = ApiError.class),
> Le 22/06/2015 19:21, jsanda a écrit :
> Do you mean that there should be no response body?
> Yes. Have a look at `ApiUtils#noContent`. Actually, it behaves like this since the beginning. I think it's the right thing to do, it's not an error, it's absence of result.
> —
> Reply to this email directly or view it on GitHub <https://github.com/hawkular/hawkular-metrics/pull/270/files#r32967276>.
>